### PR TITLE
Count SiLU and GELU as real work instead of reporting them as free

### PR DESCRIPTION
**Pull request by raimbekovm**
## summary

registers silu and gelu with a rule that counts their per-element gate multiply (gelu's tanh approximation counts three, since it cubes the input inside the gate before the outer multiply), and registers the other stateless activations that have no data-multiply in their own formula — only a compare, a select, or a multiply by a fixed hyperparameter scalar — as zero-cost, matching the existing leakyrelu decision. mish, glu and hardswish have the identical genuine-multiply shape but no consumer in ultralytics/nn/ to motivate carrying a formula, and nn.fold's overlap-count formula is nontrivial for the same reason, so all four stay unregistered and warned about rather than getting a formula nobody calls.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📈 Updates THOP activation profiling and refreshes YOLO model MACs benchmarks for more accurate computation estimates.

### 📊 Key Changes
- Added operation counting for `nn.SiLU` and `nn.GELU`, including special handling for GELU’s tanh approximation.
- Registered several additional activations—including ELU, SELU, Sigmoid, Tanh, Softplus, and related layers—as zero-operation layers.
- Updated English and Chinese README benchmark tables with recalculated MAC values for YOLOv8, YOLO11, and YOLO26 models.
- Kept unsupported computational layers such as Mish, GLU, and Hardswish unregistered so THOP continues to warn users rather than report potentially inaccurate estimates.

### 🎯 Purpose & Impact
- ✅ Produces more realistic profiling results for models that use SiLU or GELU activations.
- 📊 Aligns documented YOLO benchmark values with the updated counting behavior.
- 🚀 Helps developers make better-informed decisions about model complexity, performance, and deployment resources.
- 🌍 Keeps profiling documentation consistent across English and Chinese users.

### CODE REVIEW

**Review by UltralyticsAssistant — DISMISSED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The new hook is wired into the built-in registry correctly, handles the absence of `GELU.approximate` on the project’s older supported PyTorch versions, and preserves custom-op override behavior. The per-output-element accounting is consistent with the repository’s MAC-oriented convention.

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The new hooks are registered through the existing dispatch path, count output elements consistently with the repository’s MAC convention, and handle GELU’s documented tanh approximation separately. The README result updates align with the added activation counts.

**Comment by glenn-jocher** [comment]
Closing. This deletes the documented rationale in order to install an unstated new counting convention, and that convention contradicts a rule already in the file.

## 1. BLOCKING — it charges a sigmoid zero while charging a softmax for the identical arithmetic

`calculate_softmax` bills `total_exp + total_add + total_div`. A sigmoid is `1/(1+exp(-x))` — an exp, an add and a divide. Measured on this branch, per 1000 elements:

```
nn.Softmax(dim=1)  ->     2999   (exp + add + div, per calculate_softmax)
nn.Sigmoid()       ->        0   <- the same exp + add + div, charged ZERO
nn.Softplus()      ->        0   <- log(1+exp(x)): an exp AND a log, charged ZERO
nn.Tanh()          ->        0   <- transcendental, charged ZERO
```

Same operations, 2999 versus 0. `nn.ELU`/`nn.CELU`/`nn.SELU` each evaluate an `exp` on the negative branch too, and `nn.SELU` additionally multiplies by `scale` on *every* element — so the new comment's claim, "LeakyReLU-shaped: a fixed-scalar multiply or none at all," is false for at least six of the fifteen types it introduces.

Registering those types as `zero_ops` asserts they are free. That is precisely what the comment this PR deletes warned against: *"registering one asserts it is free ... so they want formulas rather than a blanket entry."* The PR invokes that reasoning to justify adding SiLU/GELU formulas, then violates it for twelve other types, and removes the comment so the contradiction is no longer visible on the page.

## 2. BLOCKING — the stated exclusion test contradicts the PR's own inclusions

The new comment keeps Mish/GLU/Hardswish out because they have "no consumer in ultralytics/nn/". True — each has 0. But counting `nn.<name>` references under `ultralytics/nn/`:

| registered by this PR as zero_ops | consumers |
| -- | -- |
| ELU, CELU, SELU, LogSigmoid, Tanhshrink, Hardtanh, Hardsigmoid, Hardshrink, Softshrink, Softsign, RReLU, Threshold | **0 each** |

The same test that excludes three types includes twelve. And THOP is a standalone PyPI package with users well beyond this org, so "no ultralytics consumer" is the wrong test in either direction.

## 3. It makes the published numbers differently wrong, not right

Counting SiLU's `x * sigmoid(x)` multiply while omitting the sigmoid yields a figure that is neither a multiply count nor a FLOP count. Downstream, `ultralytics` doubles MACs to get FLOPs, so a SiLU would be reported as 2 FLOPs where it costs roughly an exp, an add, a divide and a multiply. Regenerating all 15 rows of both README tables on that basis moves every published figure toward a number with no clean definition.

## 4. Scope

Three unrelated changes in one PR: fifteen `zero_ops` registrations, two new counting rules, and a regeneration of the MACs tables in two languages. Any one of them is reviewable alone; together they are not.

Also, the deleted block documented why `nn.Fold` is excluded — *"it sums OVERLAPPING blocks, and an addition per input over a window is the work `count_adap_avgpool` charges"* — which is a technical reason. The replacement drops it for "no consumer in ultralytics/nn/", losing the actual explanation. That comment was load-bearing documentation, not dead code.

## What would land

If you want to revisit this, the version worth reviewing is narrow and self-consistent: pick **one** convention and apply it everywhere. Either count elementwise transcendentals (and then `nn.Sigmoid` costs what a 1-wide softmax costs, and SiLU costs sigmoid + 1), or count none of them (and then SiLU stays unregistered alongside Mish and GLU, exactly as the current comment says). Whichever you choose, it has to reconcile with `calculate_softmax` and `count_normalization` in the same PR, and the README regeneration belongs in its own.
